src/meta/src/controller/catalog/drop_op.rs

```diff
@@ -276,6 +255,49 @@ impl CatalogController {
         let (removed_source_fragments, removed_actors, removed_fragments) =
             get_fragments_for_jobs(&txn, removed_streaming_job_ids.clone()).await?;
 
+        let mut removed_sink_by_table: HashMap<TableId, Vec<SinkId>> = HashMap::new();
+        for obj in removed_objects.values() {
+            if obj.obj_type != ObjectType::Sink {
+                continue;
+            }
+            let sink = Sink::find_by_id(obj.oid)
+                .one(&txn)
+                .await?
+                .ok_or_else(|| MetaError::catalog_id_not_found("sink", obj.oid))?;
+            if let Some(target_table_id) = sink.target_table
+                && !removed_streaming_job_ids.contains(&target_table_id)
+            {
+                removed_sink_by_table
+                    .entry(target_table_id)
+                    .or_default()
+                    .push(sink.sink_id);
+            }
+        }
+
+        let mut updated_objects = Vec::with_capacity(removed_sink_by_table.len());
+        for (table_id, removed_sink_ids) in &removed_sink_by_table {
+            let pb_table =
+                update_table_incoming_sinks(&txn, *table_id, &[], removed_sink_ids).await?;
+            updated_objects.push(PbObject {
+                object_info: Some(PbObjectInfo::Table(pb_table)),
+            });
+        }
+
+        let removed_sink_fragments =
```

Comment:
Here we have to first resolve the fragment id of sink, and then resolve the downstream fragment of the sink fragment. Is it more straightforward to just resolve the Materialized fragment of the downstream table? We can query the `Fragment` table and filter by `job_id` and `fragment_type_mask`.